### PR TITLE
Move checkMissingPod func out of controller pkg.

### PR DESCRIPTION
**Pull request by wanlin31**
This commit moves the checkMissingPod function from controller
to the status package. This change allows checkMissingPod
function to stay closer with relevant content and could be used
across the packages.

### CODE REVIEW

File: status/checkMissingPods.go
Comment (codeblooded) [inline status/checkMissingPods.go:44]: nit: would you be willing to rename `relevantPods` to `ownedPods` for consistency?

File: status/checkMissingPods_test.go
Comment (codeblooded) [inline status/checkMissingPods_test.go:40]: Generally, we use `Context` closures for inner-level descriptions, not `Describe`. When we describe something, we say "this is the thing I'm testing." Context says "this is the environment or context of the thing I'm testing."

**Review by codeblooded — COMMENTED** [review]
Looking good with a couple nits for consistency. Thank you @wanlin31 for doing this when you had a spare moment. 🙂

File: status/checkMissingPods.go
Comment (codeblooded) [reply status/checkMissingPods.go:44]: This is not super important.

**Comment by codeblooded** [comment]
@wanlin31 Reviewing on my phone's GitHub app (with outage), so @paulosjca may notice some other things. -- Feel free to amend here since there is no point in committing a file and then changing the name in the next one. Since it is 1 commit and we're not developing on top of it, GitHub will mention the old commit when you force push, preserving history. If you want, you can push a new commit and squash after approval too. Either works with me 🙃

File: status/checkMissingPods.go
Comment (wanlin31) [reply status/checkMissingPods.go:44]: My bad, I change that in the controller_loadtest, forgot to change it in the separate file, will change it. :)

File: status/checkMissingPods.go
Comment (wanlin31) [reply status/checkMissingPods.go:1]: Good to learn!

**Comment by wanlin31** [comment]
> @wanlin31 Reviewing on my phone's GitHub app (with outage), so @paulosjca may notice some other things. -- Feel free to amend here since there is no point in committing a file and then changing the name in the next one. Since it is 1 commit and we're not developing on top of it, GitHub will mention the old commit when you force push, preserving history. If you want, you can push a new commit and squash after approval too. Either works with me 🙃

I agree, :)

**Review by paulosjca — APPROVED** [review]

**Review by codeblooded — APPROVED** [review]
@wanlin31 LGTM once the commit first line is shortened so GitHub doesn't cut it off 👍

**Pull request merged**
